src/transport/SecurePairingSession.cpp

```diff
@@ -0,0 +1,293 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements the CHIP SPAKE2P Session object that provides
+ *      APIs for constructing spake2p messages and establishing encryption
+ *      keys.
+ *
+ */
+
+#include <support/CodeUtils.h>
+#include <transport/SecurePairingSession.h>
+
+namespace chip {
+
+using namespace Crypto;
+
+const char * kSpake2pContext        = "CHIP 1.0 Provisioning";
+const char * kSpake2pI2RSessionInfo = "Commissioning I2R Key";
+const char * kSpake2pR2ISessionInfo = "Commissioning R2I Key";
+
+SecurePairingSession::SecurePairingSession(void) {}
+
+SecurePairingSession::~SecurePairingSession(void)
+{
+    if (mDelegate != nullptr)
+    {
+        mDelegate->Release();
+    }
+}
+
+CHIP_ERROR SecurePairingSession::Init(uint32_t setupCode, uint32_t pbkdf2IterCount, const unsigned char * salt, size_t saltLen,
+                                      SecurePairingSessionDelegate * delegate)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    VerifyOrExit(salt != NULL, err = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(saltLen > 0, err = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(delegate != NULL, err = CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
Should these two `NULL` checks actually be checking against `nullptr`?

Reply:
I think `NULL` and `nullptr`, both evaluate to 0. So functionally it should be ok
Changing it to `nullptr` will help with code clarity. I'll push an update to the PR.

Reply:
Just as an added note, there are other places within the PR where null check of points is done against `NULL`. May be worth switching them all to `nullptr` for consistency